### PR TITLE
fake ZIPCode generator

### PR DESCRIPTION
**Pull request by aharelick**
r @weronica @riyer15 @anoshaminai 
- helper methods for dealing with ZIPCodes: `ZIPCode.get_by_zip(zip_code)` and `ZIPCode.create_zip_code(zip_code)`
- `ZIPCode.generate_fake()` for creating fake ZIPCodes
- assign random zip codes to the newly created fake data in `User.set_random_zips(users, zip_codes)`

### CODE REVIEW

File: app/models/user.py [hunk 1]
```diff
@@ -180,7 +179,18 @@ def generate_fake(count=100, **kwargs):
                 db.session.rollback()
 
     @staticmethod
-    def create_confirmed_admin(first_name, last_name, email, password):
+    def set_random_zips(users, zip_codes):
```
Comment (skylerwharton) [inline app/models/user.py]: Perhaps should be `set_random_zip_codes`?

File: app/models/location.py [hunk 2]
```diff
@@ -4,10 +4,42 @@
...
-    zip_code = db.Column(db.String(5), unique=True)
+    zip_code = db.Column(db.String(5), unique=True, index=True)
     users = db.relationship('User', backref='zip_code', lazy='dynamic')
     addresses = db.relationship('Address', backref='zip_code', lazy='dynamic')
 
+    def __init__(self, zip_code):
+        self.zip_code = zip_code
+
+    @staticmethod
+    def get_by_zip(zip_code):
```
Comment (skylerwharton) [inline app/models/location.py]: Perhaps should be `get_by_zip_code` for consistency?

File: manage.py [hunk 3]
```diff
@@ -47,16 +48,20 @@ def recreate_db():
...
-def add_fake_data(number_users):
+                dest='count')
+def add_fake_data(count):
     """
     Adds fake data to the database.
     """
-    User.generate_fake(count=number_users)
+    User.generate_fake(count=count)
+    ZIPCode.generate_fake(count=count)
+    # set a random zip for each user without one
```
Comment (skylerwharton) [inline manage.py]: Nit: Comments begin with an uppercase letters and end with a period for consistency.

File: app/models/location.py [hunk 4]
```diff
@@ -4,10 +4,42 @@
 class ZIPCode(db.Model):
     __tablename__ = 'zip_codes'
     id = db.Column(db.Integer, primary_key=True)
-    zip_code = db.Column(db.String(5), unique=True)
+    zip_code = db.Column(db.String(5), unique=True, index=True)
     users = db.relationship('User', backref='zip_code', lazy='dynamic')
     addresses = db.relationship('Address', backref='zip_code', lazy='dynamic')
 
+    def __init__(self, zip_code):
```
Comment (skylerwharton) [inline app/models/location.py:11]: Should we leave a comment saying that this constructor should never be explicitly called by a user of the model? (Helper methods should be used instead.)

**Comment by skylerwharton** [comment]
I added a few comments about naming/commenting. Otherwise I think this looks great.

File: app/models/location.py (hunk 2 above)
Comment (aharelick) [inline app/models/location.py]: yeah, good idea, done.

File: app/models/user.py (hunk 1 above)
Comment (aharelick) [inline app/models/user.py]: Done.

File: manage.py (hunk 3 above)
Comment (aharelick) [inline manage.py]: Done.

File: app/models/location.py (hunk 4 above)
Comment (aharelick) [inline app/models/location.py:11]: Yeah, that's smart. Good to keep the public notified.

**Comment by aharelick** [comment]
@weronica: Made the changes you said, good stuff.
@riyer15 or @anoshaminai: Would like one of you guys to look this over before I merge, thanks.

File: app/models/location.py [hunk 5]
```diff
@@ -4,10 +4,46 @@
...
+        result = ZIPCode.get_by_zip_code(zip_code)
+        if result is None:
+            result = ZIPCode(zip_code)
+            db.session.add(result)
+            db.session.commit()
+        return result
+
+    @staticmethod
+    def generate_fake(count=10):
+        """Generate a number of fake ZIPCodes for testing."""
```
Comment (skylerwharton) [inline app/models/location.py]: Maybe should  be `"""Generate count fake ZIPCodes for testing"""`.

Comment (aharelick) [inline app/models/location.py]: Done.

**Comment by riyer15** [comment]
@aharelick it looks good to me

**Pull request merged**